vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -1073,6 +1073,7 @@ public UpdateQueue startUpdate(int sizeChange) {
         public void initialize() {
             initConnector();
             updateSelectionModeOnClient();
+            setRequestedRange(0, getPageSize());
```

Comment:
We should consider the same for `TreeGrid`.

Reply:
Good catch! Done.